website/docs/reference/widgets/menu-button.md

```diff
@@ -74,3 +74,72 @@ Each menu item can now be customized through a set of properties by clicking on
 :::info
 We currently use the icons from [Blueprint](https://blueprintjs.com) library. You can see the list of icons [here](https://blueprintjs.com/docs/#icons).
 :::
+
+# Dynamic Menu Items
+
+Previously, the Menu Button Widget only allowed users to add menu items using the UI in the property pane. They would simple click on "Add a new menu item" button and then go on to configure this individual menu item using the settings icon after the menu item.
+
+This prevented super users to add the menu options dynamically, using JS.
+
+With [#17652](https://github.com/appsmithorg/appsmith/pull/17652), users will be able to add upto 10 menu items using JS. They simply need to select Menu Item Source (a new property) as Dynamic, bind a source data, and then they can configure each menu item by referencing `currentItem` inside the configuration.
+
+### TL;DR
```

Comment:
@dhruvikn rather than a TL;DR section. It's better if you add a loom video, so that the team can create a new video from it and use it.